### PR TITLE
WIP: parted refactoring before busy partition fix

### PR DESCRIPTION
**Pull request by rosowiecki**
Some parted module refactoring before fixing  #252

I will keep this PR mergeable, but only for reviews and shippable tests, until actual fix is completed.

##### SUMMARY
parted:
* unit parameter and final script formatting in parted()
* align and unit parameters optional
* returns script as formatted by parted()
* parted() takes list of commands to execute

unittests:
* run_parted() to mock instead of parted()
* test_change_flag changed, no need to track mock calls

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
parted

##### ADDITIONAL INFORMATION

Goal:
* by default, all commands in script list will be executed at once
* if user requests busy table operation, script commands will be executed one by one
* 'busy partition' errors will be ignored
* additional checks on result table to confirm that partitions were changed on disk, even though kernel info was not updated

### CODE REVIEW

File: plugins/modules/system/parted.py [hunk 1]
```diff
@@ -102,6 +102,15 @@
...
     version_added: '0.2.0'
+  ignore_kernel:
+    description:
+      - By default parted fails when partition on C(device) is used by kernel and modification script might not be executed in whole.
+      - When C(ingore_kernel) = C(true) module edits table contents on C(device) even if current table is used by kernel
+      - This is achieved by multiple parted program calls and ignoring kernel re-reading warnings.
+      - You may need to use utilities like partprobe, kpartx or even reboot the system to make the kernel recognize changes in partition table.
+    type: bool
+    default: False
+    version+added: '1.0.0'
```
Comment (felixfontein) [inline plugins/modules/system/parted.py:113]:
```suggestion
    version_added: '1.0.0'
```

File: plugins/modules/system/parted.py [hunk 2]
```diff
@@ -102,6 +102,15 @@
      - Parameter optional, but see notes below about negative negative C(part_start) values.
     type: str
     version_added: '0.2.0'
+  ignore_kernel:
+    description:
+      - By default parted fails when partition on C(device) is used by kernel and modification script might not be executed in whole.
+      - When C(ingore_kernel) = C(true) module edits table contents on C(device) even if current table is used by kernel
```
Comment (felixfontein) [inline plugins/modules/system/parted.py:108]:
```suggestion
      - When I(ingore_kernel) = C(true) module edits table contents on C(device) even if current table is used by kernel.
```

**Comment by felixfontein** [comment]
This needs a changelog fragment.

**Comment by rosowiecki** [comment]
For anyone who cares: I haven't abandoned it (yet), I'm just very busy with other stuff.

**Comment by felixfontein** [comment]
@rosowiecki just pinging to check whether this is still on your radar, since it has been a longer time now :-)

**Comment by felixfontein** [comment]
@rosowiecki another ping :)

**Comment by felixfontein** [comment]
needs_info

**Comment by rosowiecki** [comment]
Tomorrow I'll try to resolve conflicts, but not much more. I begin to suspect that modern kernels don't have this issue anymore and it's not fun to fix issues from RHEL6 era... (I've got a working workaround in playbook anyway: maybe I'll post it as wont-fix resolution for #252.

**Comment by felixfontein** [comment]
@rosowiecki I'll close this since there hasn't been any progress since May. Please post that workaround. Thanks :)

**Pull request closed without merging**